syntax = "proto3";

package ozonmp.ise_car_api.v1;

import "validate/validate.proto";
import "google/api/annotations.proto";
import "google/protobuf/timestamp.proto";

option go_package = "github.com/ozonmp/ise-car-api/pkg/ise-car-api;ise_car_api";

// IseCarApiService - Service for working with cars
service IseCarApiService {
  // DescribeCarV1 - Describe a car
  rpc DescribeCarV1(DescribeCarV1Request) returns (DescribeCarV1Response) {
    option (google.api.http) = {
      get: "/v1/cars/{car_id}"
    };
  }
}

message Car {
  uint64 id  = 1;
  uint64 foo = 2;
  google.protobuf.Timestamp created = 3;
}

message DescribeCarV1Request {
  uint64 car_id = 1 [(validate.rules).uint64.gt = 0];
}

message DescribeCarV1Response {
  Car value = 1;
}
